File: pset6/mario.py
import cs50


def get_height():
    while True:
        print("Height: ")
        i = cs50.get_int()
        if i >= 0 and i <= 23:
            return i
        
def main():
    make_pyramid(get_height())
    
def make_pyramid(height):
    bl = 1
    for i in range(height):
        sp = height - bl
        print(' '*sp+'#'*bl+'  '+'#'*bl)
        bl += 1
        
if __name__ == '__main__':
    main()

File: pset1/mario.c
#include <stdio.h>
#include <cs50.h>

int get_height(void);
void make_pyramid(int height);
void spaces(int n);
void blocks(int n);

int main(void)
{
    
    int height = get_height();
    make_pyramid(height);
}

int get_height(void)
{
    while (true)
    {
        printf("Height: ");
        int i = get_int();
        if (i >= 0 && i <= 23)
        {
            return i;
        } 
    }
}

void make_pyramid(int height)
{
    int bl = 1;
    for (int i=0; i < height; i++)
    {
        int sp = height - bl;
        spaces(sp);
        blocks(bl);
        spaces(2);
        blocks(bl);
        printf("\n");
        bl ++;
    }
}

void spaces(int n)
{
    for (int i=0; i < n; i++)
        {
            printf(" ");
        }
}

void blocks(int n)
{
    for (int k=0; k < n; k++)
        {
            printf("#");
        }
}